.github/workflows/build-docker.yml

```diff
@@ -295,14 +317,39 @@ jobs:
 
       # Adapted from https://docs.docker.com/build/ci/github-actions/multi-platform/
       - name: Create manifest list and push
+        id: manifest-push
         working-directory: /tmp/digests
         # The readarray part is used to make sure the quoting and special characters are preserved on expansion (e.g. spaces)
         # The jq command expands the docker/metadata json "tags" array entry to `-t tag1 -t tag2 ...` for each tag in the array
         # The printf will expand the base image with the `<UV_BASE_IMG>@sha256:<sha256> ...` for each sha256 in the directory
         # The final command becomes `docker buildx imagetools create -t tag1 -t tag2 ... <UV_BASE_IMG>@sha256:<sha256_1> <UV_BASE_IMG>@sha256:<sha256_2> ...`
+        # The digest of the new manifest is then shared as the 'digest' output
         run: |
           readarray -t lines <<< "$DOCKER_METADATA_OUTPUT_ANNOTATIONS"; annotations=(); for line in "${lines[@]}"; do annotations+=(--annotation "$line"); done
           docker buildx imagetools create \
             "${annotations[@]}" \
             $(jq -cr '.tags | map("-t " + .) | join(" ")' <<< "$DOCKER_METADATA_OUTPUT_JSON") \
             $(printf '${{ env.UV_BASE_IMG }}@sha256:%s ' *)
+          # unfortunately "docker buildx imagetools create" does not yet have a
+          # clean way of sharing the digest of the manifest it creates, see
+          # https://github.com/docker/buildx/issues/2407. Using a separate command to
+          # retrieve it _now_ is better than doing it later though, as it is highly
+          # unlikely that the digest changes in the same job with the same local docker setup.
+          digest="$(
+            docker buildx imagetools inspect \
+              "${UV_BASE_IMG}:${DOCKER_METADATA_OUTPUT_VERSION}" \
+              --format '{{json .Manifest}}' \
+            | jq -r '.digest'
+          )"
+          echo "digest=${digest}" >> "$GITHUB_OUTPUT"
```

Comment:
I believe this can be done in a separate step instead to keep this step less-convoluted

Reply:
I was specifically avoiding a separate step, because that would pose a security risk.

In theory, given enough time, a docker image tag could have been moved from one digest to another. A signature on a manifest from a trusted entity has enormous value, and if an attacker found a way to _very quickly_ move the tag to a digest of their choice, before the signing step was being executed, they could steal that signature. This is why it is a reasonably big deal that `imagetools create` doesn't output the digest in a machine-readable form the moment it creates the manifest. However, since `imagetools inspect` is run within the same step, it is being executed against the same docker installation which already has the manifest locally cached and so changing the tag would require the attacker to have access to the runner while this step is being executed.

If we move this part to a new step, then you introduce many more opportunities for the tag to be moved; there is a time gap, the job is likely to be handled by a different runner which will have to fetch the digest from the container registry, etc. This greatly increases the attack surface to get that tag moved to a different digest. **I really don't want to increase the attack surface here**.

I'll see about updating the comments here to make this clearer.

Reply:
> the job is likely to be handled by a different runner which will have to fetch the digest from the container registry

Thanks for the explanation. In this particular case steps are still guaranteed to be executed within the same runner vm and job process, hence separating into a separate step is still worthwhile for readability. It doesn't give us much of a difference in terms of security in this case. We can reconsider this once `imagetools create` has a mechanism to return the digests.

Reply:
Change pushed.

Reply:
Thanks! Looks great